tests/queries/0_stateless/01455_shard_leaf_max_rows_bytes_to_read.sql

```diff
@@ -0,0 +1,14 @@
+DROP TABLE IF EXISTS test_local;
+DROP TABLE IF EXISTS test_distributed;
+
+CREATE TABLE test_local (date Date, value UInt32) ENGINE = MergeTree(date, date, 8192);
+CREATE TABLE test_distributed AS test_local ENGINE = Distributed(test_shard_localhost, currentDatabase(), test_local, rand());
+
+INSERT INTO test_local SELECT '2000-08-01', number as value from numbers(50000);
+
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read_leaf = 40000; -- { serverError 158 }
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_bytes_to_read_leaf = 40000; -- { serverError 307 }
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read_leaf = 60000;
```

Comment:
Can you also add a test for `max_bytes_to_read` (since your code touches it a little bit, and there is only one test that uses `max_bytes_to_read` - 00091_prewhere_two_conditions), smth like below should work I guess:

```sql
SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read = 60000; -- { serverError 307 }

```

Reply:
Added! Thanks!